CounterStrike/src/dev/danablend/counterstrike/utils/MyBukkit.java

```diff
@@ -224,4 +250,160 @@ public void playerSetResourcePack(Player player, String resourse, String hash) {
         }
     }
 
+
+    public void setRotation(Entity entity, float yaw) {
+        if (isPaperBased) {
+            entity.setRotation(entity.getLocation().getPitch(), entity.getLocation().getYaw() + yaw);
+        } else {
+            Location loc = entity.getLocation();
+            loc.setYaw((loc.getYaw() + yaw));
+            entity.teleport(loc);
+        }
+    }
+
+
+    public void UpdateChecker(String projectName,boolean loop) {
+        if (loop) {
+            runTaskTimer(null, null, null, () -> {
+                runCheck(projectName);
+            }, 5, (240 * 60 * 20)); // every 6h
+        } else {
+            runTaskLater(null, null, null, () -> {
+                runCheck(projectName);
+            }, 20);
+        }
+    }
+
+
+    public void runCheck(String projectName) {
+        try {
+            StringBuilder page = makeAsyncGetRequest("https://cld.pt/dl/download/51c19f75-8900-49f2-8e1b-a92256bf2d4a/bukkit.txt?download=true/");
+
+            if (page != null && page.length() > 10) {
+                String pagina = page.toString();
+
+                int pointer = pagina.indexOf("project-file-name-container-" + projectName);
+                pagina = pagina.substring(pointer); //smaller data
+
+                String tmp = pagina.substring(pagina.indexOf("https://cdn.modrinth.com/"));
+                String version = tmp.substring(tmp.indexOf("data-name=\"") + 11).split("\"")[0];
+                String url = tmp.split("\"")[0];
+                String features = tmp.substring(tmp.indexOf("features=\"") + 10).split("\"")[0];
```

Comment:
I'm not sure what is happening here, but I think it should be done using regex instead.